src/relay/pass/mac_count.cc

```diff
@@ -0,0 +1,138 @@
+/*!
+ * Copyright (c) 2019 by Contributors
+ *
+ * \file mac_count.cc
+ * \brief Pass to roughly count the number of MACs (Multiply-Accumulate) 
+ * operations of a model. Only MACs in CONV and Dense ops are counted.
+ * This pass is valid after the type infer pass is called,
+ * otherwise the count is 0.
+ */
+
+#include <tvm/relay/op.h>
+#include <tvm/relay/attrs/nn.h>
+#include <tvm/relay/expr_functor.h>
+
+namespace tvm {
+namespace relay {
+
+namespace {
+
+bool IsConv2DNode(const ExprNode* node) {
+  const auto* call_node = dynamic_cast<const CallNode*>(node);
+  return call_node != nullptr && call_node->attrs.as<Conv2DAttrs>();
+}
+
+bool IsDenseNode(const ExprNode* node) {
+  const auto* call_node = dynamic_cast<const CallNode*>(node);
+  return call_node != nullptr && call_node->attrs.as<DenseAttrs>();
+}
+
+}  // namespace
+
+class MacCounter : private ExprVisitor {
+ public:
+  MacCounter() {
+      count_ = 0;
+  }
+  static int64_t GetTotalMacNumber(const Expr& expr) {
+    MacCounter counter;
+    counter(expr);
+    return counter.count_;
+  }
+
+ private:
+  void VisitExpr_(const CallNode* call_node) final {
+    if (IsConv2DNode(call_node)) {
+      count_ += ComputeConv2DMacs(call_node);
+    } else if (IsDenseNode(call_node)) {
+      count_ += ComputeDenseMacs(call_node);
+    }
```

Comment:
should we log somewhere to remind users the pass only calculates MAC for Conv2D and Dense?

Reply:
Added a log info in the beginning of the pass.